Src/WindowsServer/WindowsServer.Shared/AzureInstanceMetadataTelemetryInitializer.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+    /// <summary>
+    /// A telemetry initializer that adds Azure specific context information to the heartbeat, if it is available.
+    /// </summary>
+    public class AzureInstanceMetadataTelemetryInitializer : ITelemetryInitializer
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureInstanceMetadataTelemetryInitializer" /> class.
+        /// Creates a heartbeat property collector that obtains and inserts data from the Azure Instance
+        /// Metadata service if it is present and available to the currently running process. If it is not
+        /// present no added IMS data is added to the heartbeat.
+        /// </summary>
+        public AzureInstanceMetadataTelemetryInitializer()
+        {
+            WindowsServerEventSource.Log.TelemetryInitializerLoaded(this.GetType().FullName);
+
+            var telemetryModules = TelemetryModules.Instance;
+            foreach (var module in telemetryModules.Modules)
+
+            {
+                if (module is IHeartbeatPropertyManager)
+                {
+                    var hbeatManager = (IHeartbeatPropertyManager)module;
```

Comment:
There's a C#6 'pattern matching' syntax for doing the "is" and "cast" in one line:
```C#
if (module is IHeartbeatPropertyManager hbeatManager)
```

Reply:
Yes, does our minimum .NET SDK (4.5) support this? When I wrote this originally I got compiler warnings for v4.0 not supporting it...

Reply:
Changed it, I'll let the build tell me if it's supported now or not.